File: boat-bay-frontend/src/app/models/boat-capability.ts
import { BoatStatistics } from './boat-dashboard';
import { BoatLintReport } from "./boat-lint-report";

export interface BoatCapability {
  id: number;
  key: string;
  name: string;
  content?: string;

  createdOn: Date;
  createdBy: String;

  lastLintReport?: BoatLintReport;

  statistics: BoatStatistics;
}

File: boat-bay-server/src/main/java/com/backbase/oss/boat/bay/web/views/dashboard/models/BoatCapability.java
package com.backbase.oss.boat.bay.web.views.dashboard.models;

import com.backbase.oss.boat.bay.service.statistics.BoatStatistics;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;

@Data
public class BoatCapability {

    private String id;
    private String key;
    private String name;
    private String content;

    private LocalDateTime createdOn;
    private String createdBy;

    private List<BoatService> services;

    private BoatLintReport lastLintReport;

    private BoatStatistics statistics;

}
